dev/ProgressBar/ProgressBar.cpp

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ProgressBar.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+ProgressBar::ProgressBar()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_ProgressBar);
+
+    SetDefaultStyleKey(this);
+
+    // NOTE: This is necessary only because Value isn't one of OUR properties, it's implemented in RangeBase.
+    // If it was one of ProgressBar's properties, defined in the IDL, you'd do it differently (see IsIndeterminate).
+    RegisterPropertyChangedCallback(winrt::RangeBase::ValueProperty(), { this, &ProgressBar::OnRangeBaseValueChanged });
+
+    // NOTE: You can also hook up to events on your control from here.
+    Loaded({ this, &ProgressBar::OnLoaded });
+
+    SetValue(s_TemplateSettingsProperty, winrt::make<::ProgressBarTemplateSettings>());
+}
+
+void ProgressBar::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    // NOTE: Example of how named parts are loaded from the template. Important to remember that it's possible for
+    // any of them not to be found, since devs can replace the template with their own.
+    m_layoutRoot.set(GetTemplateChildT<winrt::Grid>(L"LayoutRoot", controlProtected));
+    if (auto layoutRoot = m_layoutRoot.get())
```

Comment:
Accessing a tracker_ref's value via the get() method is not free and if you can avoid it you probably should. Something like this would be better
```
if (const auto layoutRoot = GetTemplateChildT<winrt::Grid>(L"LayoutRoot", controlProtected))
{
    m_layoutRoot.set(layoutRoot);
    m_layoutRootLoadedRevoker = layoutRoot.Loaded(winrt::auto_revoke, { this, &ProgressBar::OnLayoutRootLoaded });
}
```